### PR TITLE
Localization changes, new service, refactoring

### PR DESCRIPTION
**Pull request by velddev**
1. New settings service; this is currently untested, but I didn't want to make this commit any longer and start with a clean slate.
> ⚠️ this is a deploy blocking action however

2. Miki.Localization got some buffs, I resolved some memory allocations and added iterative resource loading.

example
```cs
locale.GetString(
	new LanguageResource("miki_test_string", // contains "this is a {0}."
		new LanguageResource("entity_user")) // contains "user"
```
will return 

```
this is a user.
```
without passing locale to either resource externally from Miki's side. This code has been tested 👍 

3. All {entity}NullExceptions have been replaced obsoleted in preference of `EntityNullException<TEntity>` to avoid boilerplate exceptions in the future.

### CODE REVIEW

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 3
           

Complexity increasing per file
==============================
- src/Miki/Services/Settings/SettingsService.cs  2
- src/Miki/Services/Settings/InvalidSettingException.cs  2
- src/Miki/Modules/Internal/Routines/DatadogService.cs  1
         

Complexity decreasing per file
==============================
+ src/Miki/Extensions/LocaleExtensions.cs  -3
         

Clones removed
==============
+ src/Miki/Modules/Pasta/PastaModule.cs  -1
+ src/Miki/Modules/MarriageModule.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=5245274&bid=17191523)

File: src/Miki/Extensions/LocaleExtensions.cs [hunk 1]
```diff
@@ -1,38 +1,9 @@
 ﻿namespace Miki
```
Comment (velddev) [inline src/Miki/Extensions/LocaleExtensions.cs:1]: This has been moved to Miki.Localization

File: src/Miki/Miki.csproj [hunk 2]
```diff
@@ -45,7 +45,7 @@
     <PackageReference Include="Miki.Framework.Arguments" Version="2.3.0" />
     <PackageReference Include="Miki.Framework.Commands" Version="2.3.1" />
     <PackageReference Include="Miki.Framework.Commands.Filters" Version="2.3.0" />
-    <PackageReference Include="Miki.Functional" Version="0.1.1" />
+    <PackageReference Include="Miki.Functional" Version="0.1.3" />
```
Comment (velddev) [inline src/Miki/Miki.csproj:48]: Added support for results and Result.From(Func<>)

File: src/Miki/Modules/GeneralModule.cs [hunk 3]
```diff
@@ -538,24 +538,23 @@ public async Task InviteAsync(IContext e)
         [Command("ping", "lag")]
         public Task PingAsync(IContext e)
         {
-            return new LocalizedEmbedBuilder(e.GetLocale())
-                .WithTitle(new RawResource("Ping"))
-                .WithDescription("ping_placeholder")
-                .Build()
+            var locale = e.GetLocale();
+            return new EmbedBuilder()
```
Comment (velddev) [inline src/Miki/Modules/GeneralModule.cs:542]: LocalizedEmbedBuilder has been a failed experiment and will be removed in the next iteration of Miki.Framework

File: src/Miki/Modules/Pasta/PastaModule.cs [hunk 4]
```diff
@@ -14,6 +13,7 @@ namespace Miki.Modules
     using Miki.Framework;
     using Miki.Framework.Commands;
     using Miki.Localization;
+    using Miki.Localization;
```
Comment (velddev) [inline src/Miki/Modules/Pasta/PastaModule.cs:16]: Duplicate using

File: src/Miki/Services/Settings/SettingsService.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        public async ValueTask<Result<T>> GetAsync<T>(SettingType type, long entityId)
+            where T : Enum
+        {
+            var setting = await settingsRepository.GetAsync(entityId, (int)type);
+            if(setting == null)
+            {
+                return new Result<T>(
+                    new EntityNullException<Setting>());
+            }
+            return (T)(object)setting.Value;
```
Comment (velddev) [inline src/Miki/Services/Settings/SettingsService.cs:25]: Could be improved by using omitting boxing, but there's no easy way for now to do it.

File: src/Miki/resources/locales/eng.json [hunk 6]
```diff
@@ -123,9 +123,9 @@
   "miki_module_pasta_error_search_no_results": "No search query found. Please specify what you want to search for.",
   "miki_module_pasta_error_specify": "Please specify which pasta you would like to {0}.",
   "miki_module_pasta_error_specify_check": "check",
-  "miki_module_pasta_error_specify_delete": "delete",
-  "miki_module_pasta_error_specify_edit": "edit",
-  "miki_module_pasta_error_specify_upvote": "upvote",
+  "verb_delete": "delete",
```
Comment (velddev) [inline src/Miki/resources/locales/eng.json:126]: Not sure between either Verb or Action

**Review by velddev — COMMENTED** [review]
These are my comments lol

File: src/Miki/Modules/MarriageModule.cs [hunk 7]
```diff
@@ -375,7 +375,7 @@ public async Task ShowProposalsAsync(IContext e)
 
             var repository = e.GetService<MarriageService>();
 
-			List<UserMarriedTo> proposals = await repository.GetProposalsReceived(e.GetAuthor().Id.ToDbLong());
+			List<UserMarriedTo> proposals = await repository.GetProposalsReceivedAsync(e.GetAuthor().Id.ToDbLong());
```
Comment (Drummss) [inline src/Miki/Modules/MarriageModule.cs:378]: NIT; var

File: src/Miki/Modules/MarriageModule.cs [hunk 8]
```diff
@@ -375,7 +375,7 @@ public async Task ShowProposalsAsync(IContext e)
...
 			List<string> proposalNames = new List<string>();
```
Comment (Drummss) [inline src/Miki/Modules/MarriageModule.cs:379]: NIT; var

**Review by Drummss — APPROVED** [review]
Looks, good. I'm comfortable with this being pushed so long as testing on the SettingsModule can be done soon

File: src/Miki/Miki.csproj (hunk 2 above)
Comment (Drummss) [reply src/Miki/Miki.csproj:48]: Nice addition

File: src/Miki/Modules/GeneralModule.cs (hunk 3 above)
Comment (Drummss) [reply src/Miki/Modules/GeneralModule.cs:542]: I prefer the e.GetLocale() method :)

File: src/Miki/Modules/Pasta/PastaModule.cs (hunk 4 above)
Comment (Drummss) [reply src/Miki/Modules/Pasta/PastaModule.cs:16]: 😓 Good catch

File: src/Miki/Services/Settings/SettingsService.cs (hunk 5 above)
Comment (Drummss) [reply src/Miki/Services/Settings/SettingsService.cs:25]: Add a todo or an issue for it

File: src/Miki/resources/locales/eng.json (hunk 6 above)
Comment (Drummss) [reply src/Miki/resources/locales/eng.json:126]: Verb looks good to me, but either way, it gets the message across. I like the shortened format

**Pull request merged**